File: src/main/java/com/thingple/version/AppInstallReceiver.java
package com.thingple.version;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.util.Log;

public class AppInstallReceiver extends BroadcastReceiver {

    private static String TAG = AppInstallReceiver.class.getSimpleName();

    @Override
    public void onReceive(Context context, Intent intent) {

        Log.d(TAG, "有APP改动");
        String action = intent.getAction();
        Uri data = intent.getData();
        if (action == null || data == null) {
            return;
        }
        String packageName = data.getSchemeSpecificPart();
        Log.d(TAG, "action:" + action);
        Log.d(TAG, "package name:" + packageName);
        if (action.equals(Intent.ACTION_PACKAGE_ADDED) || action.equals(Intent.ACTION_PACKAGE_REPLACED)) {
            alertInstallComplete(packageName);
        }
    }

    private void alertInstallComplete(String packageName) {
        VersionManager versionManager = VersionManager.shareInstance();
        versionManager.appInstalled(packageName);
    }
}

File: src/main/java/com/thingple/version/AppInstallReceiver.kt
package com.thingple.version

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class AppInstallReceiver : BroadcastReceiver() {

    private val TAG = AppInstallReceiver::class.java.simpleName

    override fun onReceive(context: Context, intent: Intent) {

        Log.d(TAG, "有APP改动")
        val action = intent.action
        val data = intent.data
        if (action == null || data == null) {
            return
        }
        val packageName = data.schemeSpecificPart
        Log.d(TAG, "action:" + action)
        Log.d(TAG, "package name:" + packageName)
        if (action == Intent.ACTION_PACKAGE_ADDED || action == Intent.ACTION_PACKAGE_REPLACED) {
            alertInstallComplete(packageName)
        }
    }

    private fun alertInstallComplete(packageName: String) {
        val versionManager = VersionManager.shareInstance()
        versionManager.appInstalled(packageName)
    }
}
